fe/src/main/java/org/apache/doris/analysis/IndexDef.java

```diff
@@ -0,0 +1,165 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.analysis;
+
+import org.apache.doris.catalog.AggregateType;
+import org.apache.doris.catalog.Column;
+import org.apache.doris.catalog.PrimitiveType;
+import org.apache.doris.common.AnalysisException;
+
+import org.apache.commons.lang.StringUtils;
+
+import java.util.List;
+import java.util.TreeSet;
+
+public class IndexDef {
+    private String indexName;
+    private List<String> columns;
+    private IndexType indexType;
+    private String comment;
+
+    public IndexDef(String indexName, List<String> columns, IndexType indexType, String comment) {
+        this.indexName = indexName;
+        this.columns = columns;
+        if (indexType == null) {
+            this.indexType = IndexType.BITMAP;
+        } else {
+            this.indexType = indexType;
+        }
+        if (columns == null) {
+            this.comment = "";
+        } else {
+            this.comment = comment;
+        }
+    }
+
+    public void analyze(boolean isOlap, List<Column> tableColumns) throws AnalysisException {
+        if (indexType == IndexDef.IndexType.BITMAP) {
+            if (!isOlap) {
```

Comment:
This class is a common index definition, better not to check isOlap here.
Maybe Doris support other types of table which is not olap, but it supports bitmap index too.